feat(ingest): add HBase source connector plugin

btkcodedev opened the pull request:
Closes #14733

## Summary
Adds an HBase metadata ingestion plugin to DataHub to extract namespaces, tables, column families and schema metadata via the HBase Thrift API. Includes backend source implementation, plugin registration, UI config stubs, docs and minimal wiring to surface the connector in the ingestion UI.

## Motivation
Support for HBase metadata ingestion so users can index HBase namespaces/tables into DataHub and make them discoverable, searchable and lineage-able.

## What changed
Added backend source implementation
Registered plugin entrypoint

The connector is a StatefulIngestionSourceBase and supports:
Containers: HBase namespaces
Schema metadata: row key + column families / qualifiers
Deletion detection via stateful ingestion
Platform instance and env config
Thrift-based connection (Thrift / happybase recommended)

## Compatibility / Risk

No breaking changes.
Requires Thrift-related dependencies and (optionally) UI assets.
If Thrift libs missing, connector logs a clear error and fails to connect.

sgomezvillamor commented on metadata-ingestion/setup.py:
```diff
@@ -829,6 +829,7 @@
         "glue = datahub.ingestion.source.aws.glue:GlueSource",
         "sagemaker = datahub.ingestion.source.aws.sagemaker:SagemakerSource",
         "hana = datahub.ingestion.source.sql.hana:HanaSource",
+        "hbase = datahub.ingestion.source.sql.hbase:HBaseSource",
```
additional to the entry point, hbase dependencies should be set here

https://github.com/btkcodedev/datahub/blob/85649af65c0f1e3068d68633b3e983994710eea2/metadata-ingestion/setup.py#L404-L405

sgomezvillamor reviewed (changes requested):
Overall looks good

I miss:
* unit/integration tests
* some minimal user docs in https://github.com/btkcodedev/datahub/tree/85649af65c0f1e3068d68633b3e983994710eea2/metadata-ingestion/docs/sources

btkcodedev commented on datahub-web-react/src/images/hbaselogo.png:
Resized to square

sgomezvillamor commented on metadata-ingestion/src/datahub/ingestion/source/sql/hbase.py:
hbase is not an sql database
so sources should not be in `metadata-ingestion/src/datahub/ingestion/source/sql/`

may you place them in `metadata-ingestion/src/datahub/ingestion/source/hbase/`?

sgomezvillamor commented on metadata-ingestion/src/datahub/ingestion/source/sql/hbase.py:
```diff
@@ -0,0 +1,564 @@
+    )
+    auth_mechanism: Optional[str] = Field(
+        default=None,
+        description="Authentication mechanism (None, KERBEROS, or custom)",
```
do we support kerberos in the current implementation?
what does custom mean?

sgomezvillamor commented on metadata-ingestion/src/datahub/ingestion/source/sql/hbase.py:
```diff
@@ -0,0 +1,564 @@
+    - Table properties and configuration
+
+    HBase is a distributed, scalable, big data store built on top of Hadoop.
+    This connector uses the HBase Thrift API to extract metadata.
```
also, have you considered happybase? it seems a very popular python lib for consuming hbase and it may save you from setting up thirft

btkcodedev replied:
Replaced raw Thrift implementation with `happybase` library

btkcodedev commented:
- Moved from sql/ to src/datahub/ingestion/source/hbase/
- Switched to happybase library (replaced raw Thrift)
- Removed unused max_column_qualifiers config field
- Removed unsupported auth_mechanism config field
- Added proper connection cleanup in close() method
- Added 65+ test cases (20+ config, 35+ unit, 10+ integration)
- Updated setup.py dependencies: happybase>=1.2.0


All tests are passing
<img width="1124" height="869" alt="image" src="https://github.com/user-attachments/assets/3bacdaa2-97f1-4878-8a78-7cbbf3e0596f" />

btkcodedev commented:
*Integration tests are in development - (facing some issues with docker)

btkcodedev commented:
Integration tests are passing with mock

<img width="1117" height="394" alt="image" src="https://github.com/user-attachments/assets/b7b45468-3878-4b49-b49a-263099c5f472" />
